connect/cli/plugins/project/report/validations.py

```diff
@@ -0,0 +1,146 @@
+
+
+def validate_reports_json(project_dir, context):
+    messages = []
+    descriptor_file = os.path.join(project_dir, 'reports.json')
+    if not os.path.isfile(descriptor_file):
+        messages.append(
+            ValidationItem(
+                'ERROR',
+                f'The directory `{project_dir}` does not look like a report project directory,'
+                ' the mandatory `reports.json` file descriptor is not present.',
+            ),
+        )
+    else:
+        try:
+            _load_reports_json(project_dir)
+        except json.JSONDecodeError:
+            messages.append(
+                ValidationItem(
+                    'ERROR',
+                    'The report project descriptor `reports.json` is not a valid json file.',
+                ),
+            )
+    return ValidationResult(messages, True if messages else False)
+
+
+def validate_report_json_with_schema(project_dir, context):
+    messages = []
+    project_desc = _load_reports_json(project_dir)
+    errors = validate_with_schema(project_desc)
+    if errors:
+        messages.append(
+            ValidationItem(
+                level='ERROR',
+                message=errors,
+            ),
+        )
+    return ValidationResult(messages, True if messages else False)
+
+
+def validate_repository_definition(project_dir, context):
+    messages = []
+    data = _load_reports_json(project_dir)
+    report_project = parse(project_dir, data)
+    errors = validate(report_project)
+    if errors:
+        for error in errors:
+            messages.append(
+                ValidationItem(
+                    level='ERROR',
+                    message=error,
+                ),
+            )
+    else:
+        for report in report_project.reports:
+            error = _validate_entrypoint(
+                project_dir,
+                report.entrypoint,
+                report.report_spec,
+            )
+            if error:
+                messages.append(error)
+    return ValidationResult(messages, True if messages else False)
+
+
+def _load_reports_json(project_dir):
+    descriptor_file = os.path.join(project_dir, 'reports.json')
+    with open(descriptor_file, 'r') as f:
+        return json.load(f)
+
+
+def _validate_entrypoint(project_dir, entrypoint, report_spec):
+    sys.path.append(project_dir)
+    package = entrypoint.rsplit('.', 1)[0]
+    try:
+        entrypoint_module = importlib.import_module(package)
+    except ImportError as error:
+        return ValidationItem(
+            'ERROR',
+            f'\nErrors detected on entrypoint module: {error}',
+        )
+    # Function validation
+    if report_spec == '1' and len(inspect.signature(entrypoint_module.generate).parameters) != 3:
+        return ValidationItem(
+            'ERROR',
+            'Wrong arguments on `generate` function. The signature must be:'
+            '\n>> def generate(client, parameters, progress_callback) <<',
```

Comment:
Please make this error renders like for extension project.
Since the message supports markdown, 
the
```'\n>> def generate(client, parameters, progress_callback) <<',```

renders like a blockquote